bob-backend/src/pearl/group.rs

```diff
@@ -146,16 +146,18 @@ impl Group {
     // find in all pearls actual pearl
     async fn find_actual_holder(&self, data: &BobData) -> BackendResult<(ChildId, Holder)> {
         let holders = self.holders.read().await;
-        let mut holders_for_time: Vec<(usize, &Holder)> = holders
+        let mut holders_for_time = holders
             .iter()
             .enumerate()
-            .filter(|h| h.1.gets_into_interval(data.meta().timestamp()))
-            .collect();
+            .filter(|h| h.1.gets_into_interval(data.meta().timestamp()));
 
-        if !holders_for_time.is_empty() {
-            holders_for_time.sort_by_key(|h| h.1.start_timestamp());
-            let (id, holder) = holders_for_time.pop().expect("not empty");
-            Ok((id, holder.clone()))
+        if let Some(mut max) = holders_for_time.next() {
```

Comment:
Iteration performed two times: first time here when calling `next` and the second time below in for loop. It is better to do it once with just a for loop